localstack/services/s3/s3_listener.py

```diff
@@ -504,7 +510,7 @@ def return_response(self, method, path, data, headers, response):
             bucket_name_in_host or (len(path[1:].split('/')) > 1 and len(path[1:].split('/')[1]) > 0),
             # don't send notification if url has a query part (some/path/with?query)
             # (query can be one of 'notification', 'lifecycle', 'tagging', etc)
-            not parsed.query
+            not parsed.query or multipart_upload_complete
```

Comment:
nitpick: we could pull the evaluation of `multipart_upload_complete` out of the `all(...)` function, like this:
```
should_send_notifications = multipart_upload_complete or all([...])
```